src/main/java/com/openshift/internal/restclient/capability/resources/DockerRegistryImageStreamImportCapability.java

```diff
@@ -0,0 +1,276 @@
+
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.http.HttpHeader;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.jboss.dmr.ModelNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.openshift.internal.restclient.model.ModelNodeBuilder;
+import com.openshift.internal.restclient.model.image.ImageStreamImport;
+import com.openshift.internal.restclient.model.properties.ResourcePropertyKeys;
+import com.openshift.internal.util.JBossDmrExtentions;
+import com.openshift.restclient.IResourceFactory;
+import com.openshift.restclient.ResourceKind;
+import com.openshift.restclient.capability.resources.IImageStreamImportCapability;
+import com.openshift.restclient.http.IHttpConstants;
+import com.openshift.restclient.images.DockerImageURI;
+import com.openshift.restclient.model.IProject;
+import com.openshift.restclient.model.image.IImageStreamImport;
+
+/**
+ * Retrieve metadata directly from docker.
+ * @author jeff.cantrill
+ *
+ */
+public class DockerRegistryImageStreamImportCapability implements IImageStreamImportCapability, IHttpConstants, ResourcePropertyKeys {
+	
+	private static final String STATUS_STATUS = "status.status";
+	private static final int TIMEOUT = 10 * 1000; //10 seconds
+	private static final String ID = "id";
+	private static final String PARENT = "parent";
+	private static final String REALM = "realm";
+	private static final Logger LOG = LoggerFactory.getLogger(IImageStreamImportCapability.class);
+	private static final String DEFAULT_DOCKER_REGISTRY = "https://registry-1.docker.io/v2/";
+	private IResourceFactory factory;
+	private IProject project;
+
+	public DockerRegistryImageStreamImportCapability(IProject project, IResourceFactory factory) {
+		this.factory = factory;
+		this.project = project;
+	}
+	
+	@Override
+	public boolean isSupported() {
+		return true;
+	}
+
+	@Override
+	public String getName() {
+		return DockerRegistryImageStreamImportCapability.class.getSimpleName();
+	}
+	
+	private boolean registryExists(HttpClient client) throws Exception{
+		ContentResponse response = client.newRequest(DEFAULT_DOCKER_REGISTRY).send();
+		return (response.getStatus() == STATUS_UNAUTHORIZED || response.getStatus() == STATUS_OK);
+	}
+	
+	/**
+	 * @return the token required to pull docker metadata
+	 */
+	private String retrieveAuthToken(HttpClient client, String details) throws Exception {
+		if(StringUtils.isNotBlank(details)) {
+			Map<String, String> auth = parseAuthDetails(details);
+			if(auth.containsKey(REALM)) {
+				Request request = client.newRequest(StringUtils.strip(auth.get(REALM),"\""));
```

Comment:
the whole block where you create the request could be a separate method ex. createRequest?